### PR TITLE
Razzle Dazzle PID

### PR DESCRIPTION
**Pull request by etyley**
Adding header files for MVA PID and associated data products, see sbncode PR of the same name for details.

### CODE REVIEW

File: sbnobj/Common/Reco/ScatterClosestApproach.h
Comment (cjbackhouse) [inline sbnobj/Common/Reco/ScatterClosestApproach.h:23]: We use `f` for fields elsewhere. But I also think for a "struct-like" data product like this where the fields are directly accessible I just wouldn't prefix at all.

File: sbnobj/Common/Reco/MVAPID.cc
Comment (cjbackhouse) [inline sbnobj/Common/Reco/MVAPID.cc:10]: I didn't know you could do multiple assignment like this in C++ yet, cool.

That said, you don't need the iter so I guess you could write `if(!mMVAScoreMap.try_emplace(pdg, score).second){`

File: sbnobj/Common/Reco/MVAPID.cc
Comment (cjbackhouse) [inline sbnobj/Common/Reco/MVAPID.cc:21]: I suspect you can just write `return {-1, std::numeric_limits<float>::lowest())};`?

File: sbnobj/Common/Reco/MVAPID.h
Comment (cjbackhouse) [inline sbnobj/Common/Reco/MVAPID.h:3]: Capitalize include guards

File: sbnobj/Common/Reco/MVAPID.h
Comment (cjbackhouse) [inline sbnobj/Common/Reco/MVAPID.h:18]: Typo in comment. Also, we know they're functions. How about just "Return the map entry with the highest score"?

File: sbnobj/Common/Reco/MVAPID.cc
Comment (cjbackhouse) [inline sbnobj/Common/Reco/MVAPID.cc:1]: I was going to comment that implementation files should be .cxx and that .cc is for modules. But I see we have both conventions within sbnobj already :sob:

File: sbnobj/Common/Reco/MVAPID.h
Comment (cjbackhouse) [inline sbnobj/Common/Reco/MVAPID.h:7]: No need for string here

File: sbnobj/Common/Reco/StoppingChi2Fit.h
Comment (cjbackhouse) [inline sbnobj/Common/Reco/StoppingChi2Fit.h:23]: Same comment about removing the `m`s

**Review by cjbackhouse — CHANGES_REQUESTED** [review]
Fine, but I'm requesting variable name changes before they get baked into files.

File: sbnobj/Common/Reco/MVAPID.h
Comment (etyley) [reply sbnobj/Common/Reco/MVAPID.h:3]: I was just following what has been done in other files: 
```FlashTriggerPrimitive.hh:#ifndef FlashTriggerPrimitive_SBN_hh_
MergedTrackInfo.hh:#ifndef sbncode_MergedTrack_HH
MVAPID.h:#ifndef sbncode_MVAPID_H
RangeP.h:#ifndef sbncode_rangep_HH
ScatterClosestApproach.h:#ifndef sbncode_scatterclosestapproach_H
ShowerSelectionVars.h:#ifndef sbncode_Showerselectionvars_H
SimpleFlashMatchVars.h:#ifndef sbncode_Simpleflashmatchvars_H
StoppingChi2Fit.h:#ifndef sbncode_stoppingchi2fit_H
Stub.h:#ifndef sbncode_Stub_HH
VertexHit.h:#ifndef sbncode_VertexHit_HH
```

I am happy to update either mine or all of them if this is something you feel strongly on?

Comment (cjbackhouse) [reply sbnobj/Common/Reco/MVAPID.h:3]: Maybe let's make another PR that changes all the include guards and renames the cc's to cxx's? The problem will come when it turns out that someone has been enforcing the opposite convention to me...

Comment (etyley) [reply sbnobj/Common/Reco/MVAPID.h:3]: Okay, that sounds good to me. I assume you have a strong opinion on `.h` vs `.hh` which are also used interchangeably currently and should be fixed in that PR

Comment (cjbackhouse) [reply sbnobj/Common/Reco/MVAPID.h:3]: `.h` please. Although changing those are presumably going to have an impact across all sorts of downstream repositories.

Comment (etyley) [reply sbnobj/Common/Reco/MVAPID.h:3]: Yeah, that will also require PRs in sbncode (and maybe sbndcode and icaruscode). If it sounds good to you, I will wait until this PR is merged then go through and update this?

**Comment by etyley** [comment]
I think I have addressed all of your comments

File: sbnobj/Common/Reco/ScatterClosestApproach.h
Comment (cjbackhouse) [reply sbnobj/Common/Reco/ScatterClosestApproach.h]: sorry, can we lowercase the initial letters too? (I assume `max` doesn't cause trouble by clashing with a function?)

File: sbnobj/Common/Reco/MVAPID.h
Comment (cjbackhouse) [reply sbnobj/Common/Reco/MVAPID.h:3]: :+1:

File: sbnobj/Common/Reco/StoppingChi2Fit.h
Comment (cjbackhouse) [reply sbnobj/Common/Reco/StoppingChi2Fit.h]: same here

File: sbnobj/Common/Reco/ScatterClosestApproach.h
Comment (etyley) [reply sbnobj/Common/Reco/ScatterClosestApproach.h]: Done

**Review by cjbackhouse — APPROVED** [review]

**Pull request merged**
